library/cwm/src/lib/cwm/multi_status_selector.rb

```diff
@@ -0,0 +1,470 @@
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "cwm"
+require "abstract_method"
+
+module CWM
+  # Custom widget to manage multi status selection items
+  #
+  # It uses a RichText to emulate the multi selection list meeting following requirements:
+  #
+  #  - Allow to select more than one item.
+  #  - Able to represent multiple statuses: no selected, selected, auto selected.
+  #  - Items can be enable or disabled.
+  #  - Emit different events to distinguish the interaction through check box input or its label.
+  #  - Automatic text wrapping.
+  #  - Keep the vertical scroll.
+  #
+  # If you want to see it in action, have a look at yast2-registration or yast2-packager modules.
+  #
+  # @example Defining a MultiStatusSelector to manage products selection (with dependencies)
+  #
+  #     require "cwm/multi_status_selector"
+  #
+  #     class MyMultiStatusSelector < CWM::MultiStatusSelector
+  #       attr_reader :items
+  #
+  #       def initialize(products)
+  #         @products = products
+  #         @items = products.map { |p| Item.new(p) }
+  #       end
+  #
+  #       def contents
+  #         VBox(
+  #           VWeight(60, super),
+  #           VWeight(40, details)
+  #         )
+  #       end
+  #
+  #       def toggle(item)
+  #         item.toggle
+  #         select_dependencies
+  #         label_event_handler(item)
+  #       end
+  #
+  #       private
+  #
+  #       attr_accessor :products
+  #
+  #       def details
+  #         @details ||= CWM::RichText.new
+  #       end
+  #
+  #       def select_dependencies
+  #         # logic to recalculate auto selected items
+  #       end
+  #
+  #       def label_event_handler(item)
+  #         details.value = item.description
+  #       end
+  #
+  #       class Item < CWM::MultiStateSelector::Item
+  #         attr_reader :status, :dependencies
+  #
+  #         def initialize(product)
+  #           @product = product
+  #           @status = product.status || UNSELECTED
+  #           @dependencies = product.dependencies || []
+  #         end
+  #
+  #         def id
+  #           product.id
+  #         end
+  #
+  #         def label
+  #           product.friendly_name || product.name
+  #         end
+  #
+  #         def description
+  #           # build the item description
+  #         end
+  #
+  #         private
+  #
+  #         attr_reader :product
+  #       end
+  #     end
+  class MultiStatusSelector < CustomWidget
+    # @!method items
+    #   The items collection
+    #   @return [Array<Item>] the collection of available items
+    abstract_method :items
+
+    # @macro seeAbstractWidget
+    def init
+      refresh
+    end
+
+    # @macro seeAbstractWidget
+    def contents
+      HBox(content)
+    end
+
+    # @macro seeAbstractWidget
+    def handle(event)
+      if event["ID"].to_s.match?(Item.event_id)
```

Comment:
why match? and not include?

Reply:
Just a leftover, I guess. I'll use `#include?` instead. Thanks